zokrates_core/src/static_analysis/propagation.rs

```diff
@@ -1135,29 +1163,32 @@ impl<'ast, 'a, T: Field> ResultFolder<'ast, T> for Propagator<'ast, 'a, T> {
                             .unwrap()
                             .1
                         {
-                            TypedExpression::FieldElement(s) => s,
+                            TypedExpression::FieldElement(s) => Ok(s),
                             _ => unreachable!("????"),
```

Comment:
This message could be better, if it's in the scope of this PR to also make the panics clearer.